api/v1/controllers/subjects.js

```diff
@@ -215,7 +216,25 @@ module.exports = {
    */
   findSubjects(req, res, next) {
     validateTags(null, req.swagger.params);
-    doFind(req, res, next, helper);
+    if (featureToggles.isFeatureEnabled(sampleStoreConstants.featureName) &&
+      featureToggles.isFeatureEnabled('getSubjectFromCache')) {
+      const resultObj = { reqStartTime: new Date() }; // for logging
+      redisSubjectModel.findSubjects(req, res, resultObj)
+      .then((response) => {
+        // loop through remove values to delete property
+        if (helper.fieldsToExclude) {
+          for (let i = response.length - 1; i >= 0; i--) {
+            u.removeFieldsFromResponse(helper.fieldsToExclude, response[i]);
+          }
+        }
+
+        u.logAPI(req, resultObj, response); // audit log
+        res.status(httpStatus.OK).json(response);
+      })
+      .catch((err) => u.handleError(next, err, helper.modelName));
+    } else {
+      doFind(req, res, next, helper);
+    }
   },
 
```

Comment:
There is a clashing of opinions on where the cache/non-cache function splits off. @iamigo prefers to keep the cache-specific versions in the controllers file, whereas @shriramshankar prefers to push the cache vs non-cache implementation block into the helpers/verbs folder. This is not good as we are left with inconsistent code: ie. even in the same controller file, findSamples defines what to do when cache is turned on or off, whereas getSample calls doGet which then defines what to do when cache is turned on or off. It is better to decide on where to put the conditional and stick with it.